test-framework/security-jwt/src/main/java/io/quarkus/test/security/jwt/Claim.java

```diff
@@ -7,7 +7,19 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Target({})
 public @interface Claim {
+    /**
+     * Claim name
+     */
     String key();
 
+    /**
+     * Claim value
+     */
     String value();
+
+    /**
+     * Claim value type, the value will be converted to String if this type is set to Object.class.
+     * Supported types: String, Integer, int, Long, long, Boolean, boolean, jakarta.json.JsonArray, jakarta.json.JsonObject.
+     */
+    Class<?> type() default Object.class;
```

Comment:
How about using an enum for the valid types?

Reply:
Asking because UX-wise that would be easier to configure, but not sure if it makes sense to be that restricted

Reply:
Also because any other class than the existing list used here will not work

Reply:
Hi @gastaldi @geoand I've introduced a enum instead, note I still need to use a default value such as `Type.UNKNOWN`, because the idea is that for the well known claim such as `exp` (expiry), which, in `JsonWebToken` API is returned as `long`, users should not be required to add a `Type.LONG` hint. And if I set the default value to `Type.String` then it would be a bit strange to still convert `exp` to `long`...

Does the way I did it look reasonable enough now ?

Reply:
It looks like the PR needs to be rebased. It has 301 commits?